### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by efewqefwfd**
Added documentation across the entire library & added missing TypeScript typings. Also added tutorials that still need to be revised/added to.

### CODE REVIEW

File: docs/tutorials/Authentication.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+### What is Authentication on Roblox?
+Authentication is pretty simple, it's how Roblox knows who you are and what account you're using. Since Roblox uses FunCaptcha and have no official bot system, developers are forced to use account [**cookies.**](https://support.mozilla.org/en-US/kb/cookies-information-websites-store-on-your-computer) This will allow Roblox to easily identify your bot's account and execute actions.
```
Comment (Neztore) [inline docs/tutorials/Authentication.md:2]: Would be nice to have some explicit language that says that we use an **authentication token** which happens to be stored within cookies. And it links to an authentication session.

File: lib/asset/uploadModel.js [hunk 2]
```diff
@@ -5,6 +5,32 @@ const http = require('../util/http.js').func
 exports.required = ['data']
 exports.optional = ['itemOptions', 'asset', 'jar']
 
+// Docs
+/**
+ * Upload a model.
+ * @category Assets
+ * @alias uploadModel
+ * @param {Buffer} data - The model data.
```
Comment (Neztore) [inline lib/asset/uploadModel.js:13]: Buffer does not match the `uploadItem` method nor the [Typescript typings](https://github.com/suufi/noblox.js/blob/master/typings/index.d.ts#L984) - is it a buffer or a Stream?

File: lib/avatar/getCurrentAvatar.js [hunk 3]
```diff
@@ -1,7 +1,20 @@
+/* @module Avatar */
```
Comment (Neztore) [inline lib/avatar/getCurrentAvatar.js:1]: Is this meant to be here? I note none of the other avatar-related files have a `@module` decorator.

File: lib/badges/getAwardedTimestamps.js [hunk 4]
```diff
@@ -4,6 +4,18 @@ const http = require('../util/http').func
 // Args
 exports.required = ['userId', 'badgeId']
 
+// Docs
+/**
+ * Get the time the badge was awarded to a user.
+ * @category Badges
+ * @alias getAwardedTimestamps
+ * @param {number} userId - The userId of the user.
+ * @param {Array} badgeIds - The badge's id.
```
Comment (Neztore) [inline lib/badges/getAwardedTimestamps.js:13]: Pluralisation - The badge ids

**Review by Neztore — CHANGES_REQUESTED** [review]
Preliminary review - I haven't reviewed all the files but I'm going to check over the typings before finishing this.
I may fix some of these issues, or if you could that'd be great. If you object to any, lmk.

**Review by suufi — APPROVED** [review]
I've corrected any mistakes I saw and revised it so that it targets the docs/ folder for deployment.

**Review by Neztore — APPROVED** [review]

**Pull request merged**